src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,98 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private User newUser = new User();
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @BeforeEach
+    public void crateNewUser() {
+        newUser = new User("loginLogin", "password", 18);
+    }
+
+    @AfterEach
+    public void clearStorage() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerAllValid_Ok() {
+        registrationService.register(newUser);
+        User expected = storageDao.get(newUser.getLogin());
+        assertEquals(newUser, expected);
+    }
+
+    @Test
+    public void registerNullUser_notOk() {
+        assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(null));
+    }
+
+    @Test
+    void registerUserIsAlreadyInStorage_notOk() {
+        registrationService.register(newUser);

```

Comment:
We should interact with `storageDao` here because our registration service is under testing and we cannot rely on it's behaviour